### PR TITLE
Cleanup panzer Tpetra use

### LINKED ISSUE #4136: Panzer: address concerns in #4134
Address changes suggested by @mhoemmen in PR #4134

### PR DESCRIPTION
**Pull request by rppawlo**
Closes #4136

### CODE REVIEW

File: packages/panzer/adapters-stk/src/Panzer_STK_LocalMeshUtilities.cpp [hunk 1]
```diff
@@ -326,7 +326,7 @@ buildGhostedCellOneRing(const Teuchos::RCP<const Teuchos::Comm<int> > & comm,
 {
 
   PANZER_FUNC_TIME_MONITOR("panzer_stk::buildGhostedCellOneRing");
-  typedef Tpetra::CrsMatrix<int,int,GO> crs_type;
+  typedef Tpetra::CrsMatrix<int,int,GO,panzer::TpetraNodeType> crs_type;
```
Comment (mhoemmen) [inline packages/panzer/adapters-stk/src/Panzer_STK_LocalMeshUtilities.cpp]: In some places, you template on LO, but here, you use `LO=int`.  We haven't tested with `LO != int` for years, but you may want to fix this inconsistency at some point (not necessarily now).

File: packages/panzer/adapters-stk/src/Panzer_STK_LocalMeshUtilities.cpp [hunk 2]
```diff
@@ -411,13 +411,13 @@ buildGhostedCellOneRing(const Teuchos::RCP<const Teuchos::Comm<int> > & comm,
   */
 template <typename GO>
 Kokkos::DynRankView<double,PHX::Device>
-buildGhostedVertices(const Tpetra::Import<int,GO> & importer,
+buildGhostedVertices(const Tpetra::Import<int,GO,panzer::TpetraNodeType> & importer,
```
Comment (mhoemmen) [inline packages/panzer/adapters-stk/src/Panzer_STK_LocalMeshUtilities.cpp]: See above about `LO=int`.

File: packages/panzer/adapters-stk/src/Panzer_STK_LocalMeshUtilities.cpp [hunk 3]
```diff
@@ -411,13 +411,13 @@ buildGhostedCellOneRing(const Teuchos::RCP<const Teuchos::Comm<int> > & comm,
...
                      Kokkos::DynRankView<const double,PHX::Device> owned_vertices)
 {
   using Teuchos::RCP;
   using Teuchos::rcp;
 
-  typedef Tpetra::MultiVector<double,int,GO> mvec_type;
+  typedef Tpetra::MultiVector<double,int,GO,panzer::TpetraNodeType> mvec_type;
```
Comment (mhoemmen) [inline packages/panzer/adapters-stk/src/Panzer_STK_LocalMeshUtilities.cpp]: See above about `LO = int`.

File: packages/panzer/disc-fe/src/Panzer_L2Projection_impl.hpp [hunk 4]
```diff
@@ -261,7 +261,8 @@ namespace panzer {
 
     {
       PANZER_FUNC_TIME_MONITOR("Exporting of mass matrix");
-      ghostedMatrix->fillComplete();
+      auto ghostedMap = factory.getGhostedMap(0);
+      ghostedMatrix->fillComplete(ghostedMap,ghostedMap);
```
Comment (mhoemmen) [inline packages/panzer/disc-fe/src/Panzer_L2Projection_impl.hpp]: My guess is that `ghostedMap` is not one to one.  The domain Map and range Map (two arguments to fillComplete) _must_ be one to one.  My guess is that you want to use the row Map from the `ownedMatrix` as the domain and range Maps.

File: packages/panzer/adapters-stk/src/Panzer_STK_LocalMeshUtilities.cpp (hunk 1 above)
Comment (rppawlo) [reply packages/panzer/adapters-stk/src/Panzer_STK_LocalMeshUtilities.cpp]: yeah - we were supporting LO as a template parameter back in 2015/2016. Then about 2 years ago we decided LO should always be int in panzer. So newer code just uses int directly  and older code still has LO as a template parameter. We should fix this, just hasn't been high on the priority list.

File: packages/panzer/disc-fe/src/Panzer_L2Projection_impl.hpp (hunk 4 above)
Comment (rppawlo) [reply packages/panzer/disc-fe/src/Panzer_L2Projection_impl.hpp]: @mhoemmen - this is fixed. thanks for catching!

**Comment by eric-c-cyr** [comment]
Looking at this, I think that the key point is that there is only one set of LIDs for each cell. So the fact that the column and row maps are chosen the way there are is to support this requirement. I was always fuzzy about some of the functionality in the Petra world in terms of fillComplete. So I was usually conservative and more explicit then needed. But I believe the use of consistent LIDs is critical unless you want to carry around row LIDs and column LIDs for each cell for each matrix block.

**Comment by mhoemmen** [comment]
It's OK to choose the row and column Maps you want, but you have to pick one-to-one domain and range Maps.  However that works is your choice.

**Comment by rppawlo** [comment]
Ok - I just pushed fixes for the correct maps. Can one of you approve this again?

**Pull request merged**